### PR TITLE
Added Mesh isMeshable function to allow for better mesh point capabil…

### PR DESCRIPTION
**Pull request by darkdrgn2k**
Function to detect mesh able devices
Added false mesh positive on Orange Pi Zero onboard wifi
Added Set Canada Reg
Better way to detect interface from physical (incase its run again and no longer called wlan?)

### CODE REVIEW

**Comment by benhylau** [comment]
@darkdrgn2k Is this work-in-progress or should I review?

**Comment by darkdrgn2k** [comment]
This is a very small change that groups checks for mesh into a function.  I have another branch that im working on to use a config file and grow past the 2 meshes this script checks.

It also adds the option to downgrade the kernel for OPI Zero 

Changes should be good to go

**Comment by darkdrgn2k** [comment]
I been running this branch on all my opis and seems to be working fairly well

File: README.md [hunk 1]
```diff
@@ -80,6 +80,20 @@ If you are updating, run the same uninstall script, but keep all configuration f
 
 We are adding support for [Orange Pi](http://www.orangepi.org/) boards and have tested with the [Orange Pi One](https://dl.armbian.com/orangepione/nightly/) and [Orange Pi Lite](https://dl.armbian.com/orangepilite/nightly/). Instead of flashing Raspbian, start with the Armbian nightly images linked above, then follow the same installation steps as the Raspberry Pi.
 
+### Hardware Table
```
Comment (benhylau) [inline README.md]: `##`

File: README.md [hunk 2]
```diff
@@ -80,6 +80,20 @@ If you are updating, run the same uninstall script, but keep all configuration f
...
+
+Following is a list of hardware tested with the install.
+
+| Hardware                  | Base OS         | CJDNS Bench   | USB  | Ethernet| Notes            |
+| :-------------------------|:----------------|:--------------|:-----|:--------|:-----------------|
+| Raspbery Pi A+            | Rasbian Lite    |               | 1    | None    |                  |
```
Comment (benhylau) [inline README.md]: `s/Raspbery/Raspberry` `s/Raspbian/Raspbian`
same with next two lines. I know you are an Orange guy but wth :P

File: scripts/install2 [hunk 3]
```diff
@@ -39,6 +46,19 @@ echo -e "\e[1;32mStarting installation on ${BOARD_FAMILY} ${BOARD_NAME} (${BOARD
...
     sudo touch /etc/udev/rules.d/80-net-setup-link.rules
+
+    #Downgrade Kernel for re-enable XRadio in version 4.10 on Orange Pi Zero
+    if ! [ -z "`uname -a | grep 4.11`" ]; then 
+        if ! [ -z "`cat /sys/bus/mmc/devices/mmc1:0001/mmc1:0001:1/uevent |grep SDIO_ID | grep 0020:2281`" ]; then
+            read -p "Downgrade Kernel to Enable XRadio on Pi Zero (Y/n)? " -n 1 -r
+            echo ""
+            if ! [[ $REPLY =~ ^[Nn]$ ]]; then
+                echo -e "\e[1;32mDowngrading Kernel...\e[0m"
+            	wget http://apt.armbian.com/pool/main/l/linux-4.10.0-sun8i/linux-image-dev-sun8i_5.26_armhf.deb && dpkg -i linux-image-dev-sun8i_5.26_armhf.deb
```
Comment (benhylau) [inline scripts/install2]: Split into two lines and not use `&&` for readability?

File: scripts/install2 [hunk 4]
```diff
@@ -39,6 +46,19 @@ echo -e "\e[1;32mStarting installation on ${BOARD_FAMILY} ${BOARD_NAME} (${BOARD
...
     # Disable predictable network interface naming on Orange Pi
     sudo touch /etc/udev/rules.d/80-net-setup-link.rules
+
+    #Downgrade Kernel for re-enable XRadio in version 4.10 on Orange Pi Zero
+    if ! [ -z "`uname -a | grep 4.11`" ]; then 
+        if ! [ -z "`cat /sys/bus/mmc/devices/mmc1:0001/mmc1:0001:1/uevent |grep SDIO_ID | grep 0020:2281`" ]; then
+            read -p "Downgrade Kernel to Enable XRadio on Pi Zero (Y/n)? " -n 1 -r
+            echo ""
+            if ! [[ $REPLY =~ ^[Nn]$ ]]; then
+                echo -e "\e[1;32mDowngrading Kernel...\e[0m"
```
Comment (benhylau) [inline scripts/install2]: Remove extra space before echo

File: scripts/install2 [hunk 5]
```diff
@@ -39,6 +46,19 @@ echo -e "\e[1;32mStarting installation on ${BOARD_FAMILY} ${BOARD_NAME} (${BOARD
 if [[ $BOARD_FAMILY == "Orange Pi" ]]; then
     # Disable predictable network interface naming on Orange Pi
     sudo touch /etc/udev/rules.d/80-net-setup-link.rules
+
+    #Downgrade Kernel for re-enable XRadio in version 4.10 on Orange Pi Zero
```
Comment (benhylau) [inline scripts/install2]: `# Downgrade Kernel to re-enable XRadio in version 4.10 on Orange Pi Zero`

File: scripts/install2 [hunk 6]
```diff
@@ -39,6 +46,19 @@ echo -e "\e[1;32mStarting installation on ${BOARD_FAMILY} ${BOARD_NAME} (${BOARD
...
+    if ! [ -z "`uname -a | grep 4.11`" ]; then 
+        if ! [ -z "`cat /sys/bus/mmc/devices/mmc1:0001/mmc1:0001:1/uevent |grep SDIO_ID | grep 0020:2281`" ]; then
```
Comment (benhylau) [inline scripts/install2]: Space before `grep SDIO_ID`

File: scripts/mesh-point/mesh [hunk 7]
```diff
@@ -1,19 +1,37 @@
...
+
+   interface=`iw dev | grep phy#$1 -A 1 | grep Interface | awk '{print $2}'`
+
+    if  [ -z "$(iw phy phy$1 info | grep 'mesh point')" ]; then
+        result=''
+    elif [ "$(basename $(readlink /sys/class/net/$interface/device/driver))" == 'xradio_wlan' ]; then
+        result='' #Bad Match no mesh point here after all!
+    else
+       result='1'
+   fi
```
Comment (benhylau) [inline scripts/mesh-point/mesh]: Indentation

File: scripts/mesh-point/mesh [hunk 8]
```diff
@@ -1,19 +1,37 @@
...
+
+}
+
 
 set -e
 
 # Set wireless regulatory domain
 sudo iw reg set CA
 
+
```
Comment (benhylau) [inline scripts/mesh-point/mesh]: Remove extra line

File: scripts/mesh-point/mesh [hunk 9]
```diff
@@ -1,19 +1,37 @@
-#!/bin/sh
+#!/bin/bash
+
+function isMeshable {
```
Comment (benhylau) [inline scripts/mesh-point/mesh]: Add doc on what the checks are trying to accomplish. Why the `'xradio_wlan'` check -> `result=''` is unclear to me.

File: README.md (hunk 2 above)
Comment (darkdrgn2k) [reply README.md]: Copy And Paste and i think you mean s/Rasbian/Raspbian/

Corrected error and replaced with upgraded chart.

File: scripts/mesh-point/mesh (hunk 9 above)
Comment (darkdrgn2k) [reply scripts/mesh-point/mesh]: xradio reports that it is mesh point capable, however it is not.  So this forces the result to be false when dealing with the xradio_wlan driver.

**Pull request merged**